crates/eips/src/eip4844/sidecar.rs

```diff
@@ -1,17 +1,34 @@
 //! EIP-4844 sidecar type
 
+#[cfg(feature = "kzg")]
+use crate::eip4844::env_settings::EnvKzgSettings;
+#[cfg(any(test, feature = "arbitrary"))]
+use crate::eip4844::MAX_BLOBS_PER_BLOCK;
 use crate::eip4844::{
     kzg_to_versioned_hash, Blob, Bytes48, BYTES_PER_BLOB, BYTES_PER_COMMITMENT, BYTES_PER_PROOF,
 };
-use alloy_primitives::{bytes::BufMut, B256};
-use alloy_rlp::{Decodable, Encodable};
-
-#[cfg(any(test, feature = "arbitrary"))]
-use crate::eip4844::MAX_BLOBS_PER_BLOCK;
 
+#[cfg(feature = "kzg")]
+use crate::BlockNumHash;
+#[cfg(feature = "serde")]
+use alloc::string::String;
 #[cfg(not(feature = "std"))]
 use alloc::vec::Vec;
-
+#[cfg(feature = "kzg")]
+use alloy_primitives::FixedBytes;
+use alloy_primitives::{bytes::BufMut, B256};
+use alloy_rlp::{Decodable, Encodable};
+#[cfg(feature = "kzg")]
+use c_kzg::KzgProof;
+#[cfg(feature = "serde")]
+use core::str::FromStr;
+#[cfg(feature = "serde")]
+use serde::Deserialize;
+#[cfg(feature = "kzg")]
+use sha2::{Digest, Sha256};
+#[cfg(feature = "kzg")]
```

Comment:
none of these should be necessary, this makes it very hard to read.

please use full paths instead, this way no import is required